src/fname.mli

```diff
@@ -0,0 +1,5 @@
+val parse : string -> (string option, string) result
+(** [parse s] parses [s] and returns a filename or [None] if the filename
+    is equivalent to [/dev/null].
+
+    Returns [Error msg] in case of error. *)
```

Comment:
I am not sure if this is a clear API, when I get `Ok None` I am not sure I would think of `/dev/null`.

I know I started with this idea originally, but I think returning `Ok "/dev/null"` is better and then other tools on top can map "/dev/null" to whatever logic they find reasonable.

Reply:
This module is not accessible outside of the patch library so i don't think we should rather make the internal API consistent with the rest of the code which expects an option (until someone proposes something better, although i'm personally fine with the current state of things)